WordPress/gutenberg
# Improvements to use-focus-first-element and utils (dom)

## What?


This PR comes with some follow-up improvements from #37934.
- Instead of trying to detect all cases in one if conditional, export some functionality to isInsideRootBlock().
- General code cleanup, reuse, and organization.

## Why?


Good to keep our code clean.

## How?


Code is cleaned up.

## Testing Instructions





1. Open a post/page.
2. Insert some blocks with no contenteditable attribute such as a columns, group, or image block.
3. The first element within the block should still receive focus. In the case of the group block, the block inserter should not receive focus.

## Screenshots or screencast
